oat-sa/extension-tao-testqti
# feat: Ability to specify the root class for items of exported test and overwrite exported test

Tickets:
* https://oat-sa.atlassian.net/browse/PISA25-195
* https://oat-sa.atlassian.net/browse/PISA25-196

## Goal
To be able to specify items' class during test import and ability to overwrite previously exported tests with the same class uri and the same name.

## Overview
Add 2 params in importing API:
* itemClassUri
* overwriteTest

## How to test
1. Specify itemClassUri parameter (example - `https://infosign.docker.localhost/tao.rdf#i6371f842b77a37010fa3100880f983`)
2. Specify overwriteTest parameter and set it to `true`
3. Import test using `/taoQtiTest/RestQtiTests/import` or `/taoQtiTest/RestQtiTests/importDeferred` endpoint
4. Import test once again
5. Items of imported test should be created in the specified class. Previously imported test and its items should be overwritten
![Screenshot from 2022-11-14 11-50-52](https://user-images.githubusercontent.com/72205523/201616304-71ea5d37-8385-46fb-838e-5e9da3bdcd13.png)

![Screenshot from 2022-11-14 11-51-03](https://user-images.githubusercontent.com/72205523/201616319-233af838-9ce5-4166-bb58-f10ca3022d15.png)

Can be tested together with https://github.com/oat-sa/tao-translation-synchronizer/pull/13
